tests/processor.test.js

```diff
@@ -345,6 +345,119 @@ describe("processor", () => {
 					assert.strictEqual(blocks[0].filename, "0.js");
 				});
 
+				it("should parse a double-quoted filename from meta", () => {
+					const code =
+						prefix +
+						[
+							'```  js filename="abc.js"',
+							"var answer = 6 * 7;",
+							"```",
+						].join("\n");
+					const blocks = processor.preprocess(code);
+
+					assert.strictEqual(blocks.length, 1);
+					assert.strictEqual(blocks[0].filename, "abc.js");
+				});
+
+				it("should parse a single-quoted filename from meta", () => {
+					const code =
+						prefix +
+						[
+							"```  js filename='abc.js'",
+							"var answer = 6 * 7;",
+							"```",
+						].join("\n");
+					const blocks = processor.preprocess(code);
+
+					assert.strictEqual(blocks.length, 1);
+					assert.strictEqual(blocks[0].filename, "abc.js");
+				});
+
+				it("should parse a double-quoted filename in a directory from meta", () => {
+					const code =
+						prefix +
+						[
+							'```  js filename="abc/def.js"',
+							"var answer = 6 * 7;",
+							"```",
+						].join("\n");
+					const blocks = processor.preprocess(code);
+
+					assert.strictEqual(blocks.length, 1);
+					assert.strictEqual(blocks[0].filename, "abc/def.js");
+				});
+
+				it("should parse a single-quoted filename in a directory from meta", () => {
+					const code =
+						prefix +
+						[
+							"```  js filename='abc/def.js'",
+							"var answer = 6 * 7;",
+							"```",
+						].join("\n");
+					const blocks = processor.preprocess(code);
+
+					assert.strictEqual(blocks.length, 1);
+					assert.strictEqual(blocks[0].filename, "abc/def.js");
+				});
+
+				it("should parse a filename with lowercase, uppercase, slashes, and spaces", () => {
+					const code =
+						prefix +
+						[
+							"```  js filename='a/_b/C D E\tF \t G.js'",
+							"var answer = 6 * 7;",
+							"```",
+						].join("\n");
+					const blocks = processor.preprocess(code);
+
+					assert.strictEqual(blocks.length, 1);
+					assert.strictEqual(blocks[0].filename, "a/_b/C_D_E_F_G.js");
+				});
+
+				it("should parse a filename each from two meta", () => {
+					const code =
+						prefix +
+						[
+							"```  js filename='abc/def.js'",
+							"var answer = 6 * 7;",
+							"```",
+							"",
+							"```  js filename='abc/def.js'",
+							"var answer = 6 * 7;",
+							"```",
+						].join("\n");
+					const blocks = processor.preprocess(code);
+
+					assert.strictEqual(blocks.length, 2);
+					assert.strictEqual(blocks[0].filename, "abc/def.js");
+					assert.strictEqual(blocks[1].filename, "abc/def.js");
+				});
+
+				for (const [descriptor, filename] of [
```

Comment:
```suggestion
				for (const [filename, descriptor] of [
```

I think these are reversed.